src/extras/core/ShapePath.js

```diff
@@ -206,6 +206,14 @@ class ShapePath {
 		// only Holes? -> probably all Shapes with wrong orientation
 		if ( ! newShapes[ 0 ] )	return	toShapesNoHoles( subPaths );
 
+		if ( newShapeHoles.length > newShapes.length ) {
+
+			const restShapeHoles = newShapeHoles.pop();
+
+			Array.prototype.push.apply( newShapeHoles[ newShapeHoles.length - 1 ], restShapeHoles );
```

Comment:
Why not just `newShapeHoles[ newShapeHoles.length - 1 ].push( ...restShapeHoles )`?

Reply:
I find both ways hard to read 😅